File: algorithm/java/src/main/java/com/liyongquan/weeklycontest/wc302/MinOperations.java
package com.liyongquan.weeklycontest.wc302;

import java.util.Arrays;

public class MinOperations {
    public int minOperations(int[] nums, int[] numsDivide) {
        Arrays.sort(nums);
        int mx = numsDivide[0];
        for (int i = 1; i < numsDivide.length; i++) {
            mx = gcd(mx, numsDivide[i]);
        }
        for (int i = 0; i < nums.length; i++) {
            if (mx % nums[i] == 0) {
                return i;
            }
        }
        return -1;
    }

    private int gcd(int a, int b) {
        if (a < b) {
            int temp = a;
            a = b;
            b = temp;
        }
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }
}

File: algorithm/golang/weekcontest/wc466/minOperations.go
package wc466

func minOperations(nums []int) int {
	equal := true
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[0] {
			equal = false
			break
		}
	}
	if equal {
		return 0
	} else {
		return 1
	}
}
